jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsDeploymentDefinition.java

```diff
@@ -93,11 +93,11 @@ private JaxrsDeploymentDefinition(boolean showResources) {
 
     @Override
     public void registerOperations(ManagementResourceRegistration resourceRegistration) {
-        super.registerOperations(resourceRegistration);
-        resourceRegistration.registerOperationHandler(ADD_DEFINITION, JaxrsSubsystemAdd.INSTANCE);
-        resourceRegistration.registerOperationHandler(REMOVE_DEFINITION, ReloadRequiredRemoveStepHandler.INSTANCE);
-        if(showResources) {
+        if (showResources) {
             resourceRegistration.registerOperationHandler(ShowJaxrsResourcesHandler.DEFINITION, new ShowJaxrsResourcesHandler());
+        } else {
+            resourceRegistration.registerOperationHandler(ADD_DEFINITION, JaxrsSubsystemAdd.INSTANCE);
+            resourceRegistration.registerOperationHandler(REMOVE_DEFINITION, ReloadRequiredRemoveStepHandler.INSTANCE);
```

Comment:
Oh, I see. This class is doing two different things; and "JaxrsDeploymentDefinition" doesn't describe one of them.

In that case the setEntryType(OperationEntry.EntryType.PRIVATE) needs to be removed from ADD_DEFINITION and REMOVE_DEFINITION since in the non-deployment resource those ops are not private.

Better IMO would be to create a new JaxrsSubsystemDefinition and use it in the normal way for the non-deployment resource. Then this is only for the deployment resource, and it can register its child resource itself instead of leaving that to JaxrsExtension.initialize.